### PR TITLE
Fixed #33508 -- Fixed DatabaseFeatures.supports_index_column_ordering on MariaDB 10.8+.

### PR DESCRIPTION
**Pull request by Alokik-Roe**
ticket-33508

### CODE REVIEW

File: django/db/backends/mysql/features.py [hunk 1]
```diff
@@ -319,6 +319,8 @@ def supports_index_column_ordering(self):
         return (
             not self.connection.mysql_is_mariadb
             and self.connection.mysql_version >= (8, 0, 1)
+            or self.connection.mysql_is_mariadb
+            and self.connection.mysql_version >= (10, 8, 0)
```
Comment (felixxm) [inline django/db/backends/mysql/features.py:323]: Trailing zeros are unnecessary. Also, it'd be more readable to keep MySQL and MariaDB in separate branches, e.g.:
```python
if self.connection.mysql_is_mariadb:
    return self.connection.mysql_version >= (10, 8)
return self.connection.mysql_version >= (8, 0, 1)
```

Comment (Alokik-Roe) [reply django/db/backends/mysql/features.py:323]: That's solid advice, I will implement it. Thanks for reviewing. :smiley:

**Comment by felixxm** [comment]
@Alokik-Roe Thanks :+1: Welcome aboard :boat:

I'd like to check our test suite with MariaDB 10.8+ before merging this patch, so I'm waiting for an official MariaDB 10.8 docker image, see https://github.com/docker-library/official-images/pull/11864.

**Comment by grooverdan** [comment]
A test image is available on quay.io/mariadb-foundation/mariadb-devel:10.8. You're welcome to use this in your CI as they follow the latest server version in a stable way and the coming next changes in the Docker Library Official Images entry point for MariaDB. I did a blog on this a while ago - https://mariadb.org/new-service-quay-io-mariadb-foundation-mariadb-devel/.

Feedback welcome.

**Comment by felixxm** [comment]
@grooverdan Thanks :+1: I'm getting conversion errors: `decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]`. This is probably an issue with `mysqlclient` or Django. I will check it.

**Comment by felixxm** [comment]
@grooverdan One small thing found is that `VERSION()` is returned as bytes instead of string.

**Comment by felixxm** [comment]
Also, `SHOW FULL TABLES;` doesn't return anything, and `0` is not recognized as `Decimal` (see `mysqlclient` [converters](https://github.com/PyMySQL/mysqlclient/blob/2316313fc6777591494e870ba0c5da0d3dcd6dc6/MySQLdb/constants/FIELD_TYPE.py#L7)).

**Comment by grooverdan** [comment]
Thanks. Its still in RC, so I'll see what can be done to fix these before GA.

**Comment by felixxm** [comment]
> Thanks. Its still in RC, so I'll see what can be done to fix these before GA.

Just a note that I have the same issues with other images from `quay.io/mariadb-foundation/mariadb-devel`, e.g. `10.7.4-MariaDB-507084517fad72346ab631e3c7493154000df0a7` or `10.6.8-MariaDB-120480d15b764d03d5d36fb68735a1975ceb3454`.

**Comment by haidong** [comment]
@felixxm I'm curious if there are logs available for the issues you mentioned:

* `VERSION()` returns bytes instead of string
* `SHOW FULL TABLES;` doesn't return anything
* `0` is not recognized as `Decimal`

Basically looking for ways to reproduce them. I've looked through the publicly accessible portion of Jenkins logs and they are mostly clean. It'd be wonderful if you can file issues here: https://jira.mariadb.org/projects/MDEV/issues so they can be tracked and addressed.

**Comment by grooverdan** [comment]
Setup:
```
$ podman run --rm --env MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1 --name m102 -d mariadb:10.2
$ podman run --rm --env MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1 --name m107 -d mariadb:10.7
$ podman run --rm --env MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1 --name qm102 -d quay.io/mariadb-foundation/mariadb-devel:10.2
$ podman run --rm --env MARIADB_ALLOW_EMPTY_ROOT_PASSWORD=1 --name qm108 -d quay.io/mariadb-foundation/mariadb-devel:10.8
$ podman run --rm --env MYSQL_ALLOW_EMPTY_PASSWORD=1 --name my80 -d  mysql:8.0
```

```
$ podman exec -ti qm102 mysql  --column-type-info -e 'select version(),0'
Field   1:  `version()`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  latin1_swedish_ci (8)
Length:     56
Max_length: 56
Decimals:   39
Flags:      NOT_NULL 

Field   2:  `0`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONGLONG
Collation:  binary (63)
Length:     1
Max_length: 1
Decimals:   0
Flags:      NOT_NULL BINARY NUM 


+----------------------------------------------------------+---+
| version()                                                | 0 |
+----------------------------------------------------------+---+
| 10.2.44-MariaDB-647e952315740286df3cf6643732ad8deb1ff7fe | 0 |
+----------------------------------------------------------+---+

$ podman exec -ti m102 mysql  --column-type-info -e 'select version(),0'
Field   1:  `version()`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  latin1_swedish_ci (8)
Length:     38
Max_length: 38
Decimals:   39
Flags:      NOT_NULL 

Field   2:  `0`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONGLONG
Collation:  binary (63)
Length:     1
Max_length: 1
Decimals:   0
Flags:      NOT_NULL BINARY NUM 


+----------------------------------------+---+
| version()                              | 0 |
+----------------------------------------+---+
| 10.2.41-MariaDB-1:10.2.41+maria~bionic | 0 |
+----------------------------------------+---+

$ podman exec -ti my80 mysql  --column-type-info -e 'select version(),0'
Field   1:  `version()`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  latin1_swedish_ci (8)
Length:     6
Max_length: 6
Decimals:   31
Flags:      NOT_NULL 

Field   2:  `0`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONGLONG
Collation:  binary (63)
Length:     2
Max_length: 1
Decimals:   0
Flags:      NOT_NULL BINARY NUM 


+-----------+---+
| version() | 0 |
+-----------+---+
| 8.0.28    | 0 |
+-----------+---+
```

Same results in docker.io/library/mariadb:10.7 and quay.io/mariadb-foundation/mariadb-devel:10.8

`SHOW FULL TABLES` requires a database or access on the tables.
```
$ podman exec -ti qm108 mysql mysql -e 'show full tables'
+---------------------------+------------+
| Tables_in_mysql           | Table_type |
+---------------------------+------------+
| column_stats              | BASE TABLE |
| columns_priv              | BASE TABLE |
| db                        | BASE TABLE |
...
```

```
$ podman exec -ti qm108 mysql  --column-type-info -e 'create database test; use test; create table t (c decimal(3,3)); insert into t values(0); show warnings; select * from t'
Field   1:  `Level`
Org_field:  ``
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  latin1_swedish_ci (8)
Length:     7
Max_length: 0
Decimals:   39
Flags:      NOT_NULL 

Field   2:  `Code`
Org_field:  ``
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       LONG
Collation:  binary (63)
Length:     4
Max_length: 0
Decimals:   0
Flags:      NOT_NULL UNSIGNED BINARY NUM 

Field   3:  `Message`
Org_field:  ``
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       VAR_STRING
Collation:  latin1_swedish_ci (8)
Length:     512
Max_length: 0
Decimals:   39
Flags:      NOT_NULL 


Field   1:  `c`
Org_field:  `c`
Catalog:    `def`
Database:   `test`
Table:      `t`
Org_table:  `t`
Type:       NEWDECIMAL
Collation:  binary (63)
Length:     5
Max_length: 5
Decimals:   3
Flags:      NUM 


+-------+
| c     |
+-------+
| 0.000 |
+-------+
```

I can't see anything changed with this regard on the MariaDB side on this container image or in the server. More details could help.

**Comment by felixxm** [comment]
I created https://jira.mariadb.org/browse/MDEV-27933.

**Comment by dr-m** [comment]
In the commit message, it might be good to mention the changes that implemented the bulk of this in MariaDB Server 10.8.1:
MariaDB/server@358921ce32203a9a8dd277a5ba7ac177c9e79e53
MariaDB/server@a4cac0e07ad7f0909be3f66676e800e20c6b9fa8
MariaDB/server@791146b9d23e24b628a013d11ca1ff0e52504578

**Pull request merged**